tests/controllers/test_rss_controller.py

```diff
@@ -141,10 +182,56 @@ def test_try_parse_feed_with_beautiful_soup(feed_input: str, is_url: bool, enabl
 
 @pytest.mark.parametrize("enable_cache", [True, False])
 def test_enable_disable_cache(enable_cache):
-    feed_input = "https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml"
-    rss_controller = RSSController(feed_input=feed_input, enable_cache=enable_cache)
+    controller = RSSController(feed_input=test_urls, enable_cache=enable_cache)
 
     if enable_cache:
-        assert rss_controller._session
+        assert controller._session
     else:
-        assert not rss_controller._session
+        assert not controller._session
+
+
+def test_parse_feeds(mock_feed: feedparser.FeedParserDict):
+    feed_input = test_urls[0]
+    controller = RSSController(feed_input=feed_input, enable_cache=False)
+
+    with patch("morpheus.controllers.rss_controller.feedparser.parse") as mock_feedparser_parse:
+
+        mock_feedparser_parse.return_value = mock_feed
+
+        with patch.object(controller, '_try_parse_feed') as mock_try_parse_feed:
+            dataframes_generator = controller.parse_feeds()
+            next(dataframes_generator, None)
+            feed_stats: FeedStats = controller._feed_stats_dict[feed_input]
```

Comment:
Same comment here; we should not be testing against internal class members. This should probably be an interface method to retrieve feed_stats for a given input id.